pkg/scheduler/framework/framework_test.go

```diff
@@ -1935,3 +1972,343 @@ func TestUpdatePolicySnapshotStatusFrom(t *testing.T) {
+			want: []fleetv1beta1.ClusterDecision{
+				{
+					ClusterName: clusterName,
+					Selected:    true,
+					ClusterScore: &fleetv1beta1.ClusterScore{
+						TopologySpreadScore: &topologySpreadScore1,
+						AffinityScore:       &affinityScore1,
+					},
+					Reason: pickedByPolicyReason,
+				},
+				{
+					ClusterName: altClusterName,
+					Selected:    true,
+					ClusterScore: &fleetv1beta1.ClusterScore{
+						TopologySpreadScore: &topologySpreadScore2,
+						AffinityScore:       &affinityScore2,
+					},
+					Reason: pickedByPolicyReason,
+				},
+				{
+					ClusterName: anotherClusterName,
+					Selected:    false,
+					Reason:      filteredStatus.String(),
+				},
+			},
+		},
+		{
+			name:                              "with filtered clusters (count exceeding limit), small number of existing bindings",
+			maxUnselectedClusterDecisionCount: 0,
+			filtered: []*filteredClusterWithStatus{
+				{
+					cluster: &fleetv1beta1.MemberCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: anotherClusterName,
+						},
+					},
+					status: filteredStatus,
+				},
+			},
+			existing: [][]*fleetv1beta1.ClusterResourceBinding{
+				{
+					&fleetv1beta1.ClusterResourceBinding{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: bindingName,
+						},
+						Spec: fleetv1beta1.ResourceBindingSpec{
+							ClusterDecision: fleetv1beta1.ClusterDecision{
+								ClusterName: clusterName,
+								Selected:    true,
+								ClusterScore: &fleetv1beta1.ClusterScore{
+									TopologySpreadScore: &topologySpreadScore1,
+									AffinityScore:       &affinityScore1,
+								},
+								Reason: pickedByPolicyReason,
+							},
+						},
+					},
+					&fleetv1beta1.ClusterResourceBinding{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: altBindingName,
+						},
+						Spec: fleetv1beta1.ResourceBindingSpec{
+							ClusterDecision: fleetv1beta1.ClusterDecision{
+								ClusterName: altClusterName,
+								Selected:    true,
+								ClusterScore: &fleetv1beta1.ClusterScore{
+									TopologySpreadScore: &topologySpreadScore2,
+									AffinityScore:       &affinityScore2,
+								},
+								Reason: pickedByPolicyReason,
+							},
+						},
+					},
+				},
+			},
+			want: []fleetv1beta1.ClusterDecision{
+				{
+					ClusterName: clusterName,
+					Selected:    true,
+					ClusterScore: &fleetv1beta1.ClusterScore{
+						TopologySpreadScore: &topologySpreadScore1,
+						AffinityScore:       &affinityScore1,
+					},
+					Reason: pickedByPolicyReason,
+				},
+				{
+					ClusterName: altClusterName,
+					Selected:    true,
+					ClusterScore: &fleetv1beta1.ClusterScore{
+						TopologySpreadScore: &topologySpreadScore2,
+						AffinityScore:       &affinityScore2,
+					},
+					Reason: pickedByPolicyReason,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			decisions := newSchedulingDecisionsFrom(tc.maxUnselectedClusterDecisionCount, tc.filtered, tc.existing...)
```

Comment:
I don't get https://github.com/Azure/fleet/blob/4f6168f8fe0d4f85fca528d67e9a9d834e805114/pkg/scheduler/framework/frameworkutils.go#L226

IIRC, maxUnselectedClusterDecisionCount is now the max total decision instead of only the unselected

Reply:
Hi Ryan! For this first one, that function (`newSchedulingDecisionsFrom`) sometimes needs to take account of a number of binding lists, e.g., bindings that have been scheduled/bound + bindings that have just been created/updated; so for convenience it uses ... so that we don't have organize the lists into a list of lists.

Reply:
For the second comment, I will explain in the thread below, if that is OK.